Brittany Crocker/Week 01/Arrays and Functions/arrays-and-functions.js

```diff
@@ -0,0 +1,151 @@
+const reverseString = function(string){
+  splitString = string.split("");
+  reverseArray = splitString.reverse();
+  joinArray = reverseArray.join("");
+  console.log(joinArray);
+}
+reverseString("hello there");
+
+
+// 6. Write a function `findLongestWord` that takes an array of words and returns the length of the longest one.
+const findLongestWord = function(stringArray){
+  let longestWord =[0];
+  for (i=0; i < stringArray.length; i++) {
+    if (stringArray[i].length > longestWord.length) {
+      longestWord.length = stringArray[i].length;
+      console.log(stringArray[i].length);
+    } else {
+      console.log("doh!");
+    }
+}
+}
+findLongestWord(["golden", "pik", "fountain"]);
+
+
+// 7. Write a function `filterLongWords` that takes an array of words and an number `i` and returns the array of words that are longer than i.
+const filterLongWords = function(words, num){
+  for (i=0; i < words.length; i++){
+    if (words[i].length > num) {
+      console.log(words[i]);
+    }
+  }
+};
+filterLongWords(["hi", "there", "champion"], 2);
+
+
+// # Homework: The Word Guesser
+//
+// You'll create a simple word guessing game where the user gets infinite tries to guess the word (like Hangman without the hangman, or like Wheel of Fortune without the wheel and fortune).
+//
+// - Create two global arrays: one to hold the letters of the word (e.g. 'F', 'O', 'X'), and one to hold the current guessed letters (e.g. it would start with '\_', '\_', '\_' and end with 'F', 'O', 'X').
+// - Write a function called guessLetter that will:
+// - Take one argument, the guessed letter.
+// - Iterate through the word letters and see if the guessed letter is in there.
+// - If the guessed letter matches a word letter, changed the guessed letters array to reflect that.
+// - When it's done iterating, it should log the current guessed letters ('F__')
+// and congratulate the user if they found a new letter.
+// - It should also figure out if there are any more letters that need to be guessed,
+// and if not, it should congratulate the user for winning the game.
+// - Pretend you don't know the word, and call guessLetter multiple times with various letters to check that your program works.
+//
+
+let word = "hip"
+word.split()
+let guessedLetters = []
+answer = []
+
+const guessLetter = function(guessedLetter){
+  for (i = 0; i <= word.length; i++){
+    if (answer.length === word.length ) {   // not working for some reason
+      console.log("Congratulations you finished the game.");
+      break;
+    }
+    else if (word.includes(guessedLetter)) {
+      let place = word.indexOf(guessedLetter);
+        answer[place] = guessedLetter;
+          console.log(`Congratulations your guess is correct! Answer = ${answer} `)
+      break;
+    }
+    else {
+      guessedLetters.push(guessedLetter);
+        console.log(`Guessed Letters = ${guessedLetters}`);
+        console.log("Try again!");
+      break;
+    }
+  }
+}
```

Comment:
Using a return statement will stop the execution of a program, so is another way to "exit" a program and also return something (like a string that says ""Congratulations you finished the game message").